### PR TITLE
[ACS-4460] add no hostname verification remove com.google.collections…

### PR DESCRIPTION
**Pull request by kcichonczyk**
… dependencies

### CODE REVIEW

File: engines/base/pom.xml [hunk 1]
```diff
@@ -56,6 +56,11 @@
             </exclusions>
         </dependency>
 
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+        </dependency>
+
```
Comment (mstrankowski) [inline engines/base/pom.xml:63]: Why the switch from collections to guava?

**Review by pzhyland — APPROVED** [review]

File: engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java [hunk 2]
```diff
@@ -164,7 +173,10 @@ private HttpClient createHttpClientWithSslContext(SslContextBuilder sslContextBu
     private RestTemplate createRestTemplateWithSslContext(SSLContextBuilder sslContextBuilder) throws NoSuchAlgorithmException, KeyManagementException {
         SSLContext sslContext = sslContextBuilder.build();
         SSLConnectionSocketFactory sslContextFactory = new SSLConnectionSocketFactory(sslContext);
-        CloseableHttpClient httpClient = HttpClients.custom().setSSLSocketFactory(sslContextFactory).build();
+
+        HttpClientBuilder httpClientBuilder = HttpClients.custom().setSSLSocketFactory(sslContextFactory);
+        if(hostNameVerificationDisabled) httpClientBuilder.setSSLHostnameVerifier(new NoopHostnameVerifier());
```
Comment (mstrankowski) [inline engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java:178]: That's great, but we also need to take a look at what Domenico provided - the security vulnerability

"If the Veracode finding actually makes sense, you guys could try customizing the Netty secure HTTP client creation similarly to how the reactor people do, and then agree with Ciro to mitigate the vulnerability as dealt with. E.g.:
    private HttpClient createHttpClientWithSslContext(SslContextBuilder sslContextBuilder) throws SSLException {
        SslContext sslContext = sslContextBuilder.build();
        return HttpClient.create().secure(p -> p.sslContext(sslContext).handlerConfigurator(handler -> {
            SSLEngine engine = handler.engine();
            SSLParameters sslParameters = engine.getSSLParameters();
            sslParameters.setEndpointIdentificationAlgorithm("HTTPS");
            engine.setSSLParameters(sslParameters);
        }));
    }"
    
Which means that we should add a hostname verification manually by default when we use a Netty httpClient directly

File: engines/base/pom.xml (hunk 1 above)
Comment (kcichonczyk) [reply engines/base/pom.xml:63]: guava contains collections already, 1.0.0 was very old version, caused problems in other projects (ai-renditions)

File: engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java (hunk 2 above)
Comment (kcichonczyk) [reply engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java:185]: updated

File: engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java [hunk 3]
```diff
@@ -132,7 +140,10 @@ public SslContextBuilder nettySslContextBuilder() throws UnrecoverableKeyExcepti
             sslContextBuilder.keyManager(keyManagerFactory);
         }
 
-        if(isTruststoreConfigured())
+        if(hostNameVerificationDisabled)
+        {
+            sslContextBuilder.trustManager(InsecureTrustManagerFactory.INSTANCE);
```
Comment (mstrankowski) [inline engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java:145]: https://stackoverflow.com/questions/64190476/what-is-the-equivalent-of-setsslhostnameverifiernoophostnameverifier-instance
InsecureTrustManagerFactory doesn't restrict itself to hostnames only, it disregards all security.

Comment (mstrankowski) [reply engines/base/src/main/java/org/alfresco/transform/base/config/MTLSConfig.java:145]: And can't you just add an if at the "createHttpClientWithSslContext" method then?

**Review by mstrankowski — APPROVED** [review]

**Pull request merged**